File: examples/javascript/examples/basic-usage.js
/**
 * Basic usage examples for Triggers API client
 */

const TriggersAPIClient = require('../src/client');

// Initialize client
// For local development:
const client = new TriggersAPIClient({
    apiKey: 'test-api-key-12345',
    baseUrl: 'http://localhost:8080'
});

// For production:
// const client = new TriggersAPIClient({
//     apiKey: 'your-production-api-key',
//     baseUrl: 'https://your-api-url.com'
// });

async function main() {
    console.log('=== Creating an Event ===');
    // Create an event
    const event = await client.createEvent({
        source: 'my-app',
        eventType: 'user.created',
        payload: { userId: '123', name: 'John Doe', email: 'john@example.com' }
    });
    console.log(`Event created: ${event.event_id}`);
    console.log(`Status: ${event.status}`);
    console.log(`Created at: ${event.created_at}`);

    console.log('\n=== Getting Event Details ===');
    // Get event details
    const eventDetails = await client.getEvent(event.event_id);
    console.log(`Event ID: ${eventDetails.event_id}`);
    console.log(`Source: ${eventDetails.source}`);
    console.log(`Event Type: ${eventDetails.event_type}`);
    console.log(`Payload:`, JSON.stringify(eventDetails.payload, null, 2));
    console.log(`Status: ${eventDetails.status}`);

    console.log('\n=== Getting Inbox ===');
    // Get pending events
    const inbox = await client.getInbox({ limit: 10 });
    console.log(`Found ${inbox.events.length} pending events`);
    for (const event of inbox.events) {
        console.log(`  - ${event.event_id}: ${event.source}/${event.event_type}`);
    }

    console.log('\n=== Acknowledging an Event ===');
    // Acknowledge the event we created
    if (inbox.events.length > 0) {
        const firstEventId = inbox.events[0].event_id;
        const ack = await client.acknowledgeEvent(firstEventId);
        console.log(`Event ${ack.event_id} acknowledged`);
        console.log(`Acknowledged at: ${ack.acknowledged_at}`);
    }

    console.log('\n=== Deleting an Event ===');
    // Delete an event
    if (inbox.events.length > 0) {
        const eventToDelete = inbox.events[0].event_id;
        const result = await client.deleteEvent(eventToDelete);
        console.log(`Event ${result.event_id} deleted: ${result.message}`);
    }

    console.log('\n=== Example Complete ===');
}

main().catch(error => {
    console.error('Error:', error.message);
    process.exit(1);
});

File: examples/python/examples/basic_usage.py
"""Basic usage examples for Triggers API client"""

from triggers_api import TriggersAPIClient

# Initialize client
# For local development:
client = TriggersAPIClient(
    api_key="test-api-key-12345",
    base_url="http://localhost:8080"
)

# For production:
# client = TriggersAPIClient(
#     api_key="your-production-api-key",
#     base_url="https://your-api-url.com"
# )

print("=== Creating an Event ===")
# Create an event
event = client.create_event(
    source="my-app",
    event_type="user.created",
    payload={"user_id": "123", "name": "John Doe", "email": "john@example.com"}
)
print(f"Event created: {event.event_id}")
print(f"Status: {event.status}")
print(f"Created at: {event.created_at}")

print("\n=== Getting Event Details ===")
# Get event details
event_details = client.get_event(event.event_id)
print(f"Event ID: {event_details.event_id}")
print(f"Source: {event_details.source}")
print(f"Event Type: {event_details.event_type}")
print(f"Payload: {event_details.payload}")
print(f"Status: {event_details.status}")

print("\n=== Getting Inbox ===")
# Get pending events
inbox = client.get_inbox(limit=10)
print(f"Found {len(inbox.events)} pending events")
for event in inbox.events:
    print(f"  - {event['event_id']}: {event['source']}/{event['event_type']}")

print("\n=== Acknowledging an Event ===")
# Acknowledge the event we created
if inbox.events:
    first_event_id = inbox.events[0]['event_id']
    ack = client.acknowledge_event(first_event_id)
    print(f"Event {ack.event_id} acknowledged")
    print(f"Acknowledged at: {ack.acknowledged_at}")

print("\n=== Deleting an Event ===")
# Delete an event
if inbox.events:
    event_to_delete = inbox.events[0]['event_id']
    result = client.delete_event(event_to_delete)
    print(f"Event {result.event_id} deleted: {result.message}")

print("\n=== Example Complete ===")


